### PR TITLE
Add ADR for MDC migration

### PR DESCRIPTION
**Pull request by msasikanth**

### CODE REVIEW

File: doc/mdc-migration.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# MDC Android migration
```
Comment (sanchita) [inline doc/mdc-migration.md:1]: Can we also not use the `MDC` acronym unless we have expanded on it in one place atleast?

Comment (msasikanth) [reply doc/mdc-migration.md:1]: I expanded it in the context part, but you're right I will use the full name in the title it self.

File: doc/mdc-migration.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Status:
+Accepted on 2020-04-16
+
+## Context:
+We have been using app compat design theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
```
Comment (sanchita) [inline doc/mdc-migration.md:6]: Can we replace the word "app compat" with `Appcompat` since it's a defined package in Android.

File: doc/mdc-migration.md (hunk 1 above)
Comment (msasikanth) [inline doc/mdc-migration.md:1]:
```suggestion
# Material Design Components Migration
```

Is this fine @sanchita ? 🤔

Comment (sanchita) [reply doc/mdc-migration.md:1]: Sure. If you will be reusing this as an acronym anywhere in this doc then you might wanted to do this:
"Material Design Components (MDC)"

File: doc/mdc-migration.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+- Primary reason why we are moving to Material Design Components theme is to use `MaterialButton`,  and avoid using different buttons such as `Button` , `OmegaCenterIconButton` or `PrimarySolidButton`.  The material theme migration will also allows us to use other material components. You can find more components supported by Android and how to theme them [here](https://material.io/components).
+- This migration will also open better theming support for material components such as corner radius, strokes, shapes etc.,
+- You can find more information on setting up MDC Android [here](https://material.io/develop/android/docs/getting-started/).
```
Comment (sanchita) [inline doc/mdc-migration.md:10]: The way we write `Context` is that we give out a list of problems we have in our current implementation or things that need improvement. 
I see that the first two points are reasons for using MDC but maybe this can be reworded to say what are the problems with using `Appcompat`. 
Also, the third is a reference and not a reason. Can we move the third point out to not being a point in the reasons list?

File: doc/mdc-migration.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+## Changes:
+### Button tag change:
+- One of the primary change with this migration is using `Button` tag, we will start using `com.google.android.material.button.MaterialButton` in views instead of `Button`, `OmegaCenterIconButton`, `PrimarySolidButton`.
+- In order to use icon in a `MaterialButton` use the `icon` attribute instead of `drawableStart` or `drawableEnd` attr.
+- `MaterialButton` also allows us to set `iconGravity` (`textStart`, `viewStart`, `textEnd`, `viewEnd`) for setting the icon position, `iconTint` for setting the icon color.
+  - By default `Clinic.V2.MaterialButton` & other primary button styles for Simple use `iconGravity` of `textStart`, if you want to change that create a new style or override it in the layout file.
+- Material theme will automatically convert any `Button` to `MaterialButton` internally using custom inflater. So if there is a view that extends button the theme values may need to changed if the design specs are broken.
+
```
Comment (sanchita) [inline doc/mdc-migration.md:19]: We need to derive the gist of how moving to this will improve our code quality and ease of use. We do not need the implementation details as much in an ADR.

File: doc/mdc-migration.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	- Outline Button: `?attr/materialButtonOutlinedStyle` or `Clinic.V2.OutlineButton`
+	- Un-Elevated/Flat Button: `@style/Clinic.Button.Flat`
+- You can extend any of those primary button styles to override certain attrs like color, for example:
+```
+<style name="Clinic.V2.MaterialButton.Green3">
+  <item name="backgroundTint">@color/green3</item>
+  <item name="android:textColor">@color/green1</item>
+  <item name="iconTint">@color/green1</item>
+</style>
+```
```
Comment (sanchita) [inline doc/mdc-migration.md:48]: Look at the previous comment to change this section.

File: doc/mdc-migration.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+Accepted on 2020-04-16
+
+## Context:
+We have been using app compat design theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
+
+- Primary reason why we are moving to Material Design Components theme is to use `MaterialButton`,  and avoid using different buttons such as `Button` , `OmegaCenterIconButton` or `PrimarySolidButton`.  The material theme migration will also allows us to use other material components. You can find more components supported by Android and how to theme them [here](https://material.io/components).
+- This migration will also open better theming support for material components such as corner radius, strokes, shapes etc.,
+- You can find more information on setting up MDC Android [here](https://material.io/develop/android/docs/getting-started/).
+
+## Changes:
```
Comment (sanchita) [inline doc/mdc-migration.md:12]: The whole section around `Changes` needs to be replaced with the `Decision` which was to move to MDC.
Please refer other ADRs to see how to write this.

File: doc/mdc-migration.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
+# MDC Android migration
+## Status:
+Accepted on 2020-04-16
+
+## Context:
+We have been using app compat design theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
+
+- Primary reason why we are moving to Material Design Components theme is to use `MaterialButton`,  and avoid using different buttons such as `Button` , `OmegaCenterIconButton` or `PrimarySolidButton`.  The material theme migration will also allows us to use other material components. You can find more components supported by Android and how to theme them [here](https://material.io/components).
```
Comment (sanchita) [inline doc/mdc-migration.md:8]: Correction:
"The material theme migration will also **allow** us to use other material components."

File: doc/mdc-migration.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  <item name="android:textColor">@color/green1</item>
+  <item name="iconTint">@color/green1</item>
+</style>
+```
+
+### Removed widgets:
+- This change will also remove `PrimarySolidButton` & `OmegaCenterIconButton`, we will be using `MaterialButton` in place of that. `MaterialButton` will give us much more control over theming and icon.
+
+## Consequences:
+- Since we started with a design theme, some of the components styles maybe completely or partially broken due this theme switch.
```
Comment (sanchita) [inline doc/mdc-migration.md:54]: Can you please expand on this point or reword it to say:
"This is a new design theme that we will be moving to so some of the component styles can be broken either partially or completely because of this theme switch. "

File: doc/mdc-migration.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  <item name="backgroundTint">@color/green3</item>
+  <item name="android:textColor">@color/green1</item>
+  <item name="iconTint">@color/green1</item>
+</style>
+```
+
+### Removed widgets:
+- This change will also remove `PrimarySolidButton` & `OmegaCenterIconButton`, we will be using `MaterialButton` in place of that. `MaterialButton` will give us much more control over theming and icon.
+
+## Consequences:
```
Comment (sanchita) [inline doc/mdc-migration.md:53]: Please think of more consequences and elaborate on them a little.
This is a very important section of an ADR.

File: doc/mdc-migration.md (hunk 2 above)
Comment (msasikanth) [reply doc/mdc-migration.md:6]: I have replaced it with `Theme.Design.*` because there are 2 variants of themes in the appcompat package. So I made specific to what we used before.

File: doc/mdc-migration.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
+# Material Design Components (MDC) Migration
+## Status:
+Accepted on 2020-04-16
+
+## Context:
+We have been using `Theme.Design.*` theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
+
```
Comment (sanchita) [inline doc/mdc-migration.md:7]: Trailing `,`. Please remove.
Also add a line saying that the following are the problems we are facing with our current theme.

File: doc/mdc-migration.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+- Right now we are using 3 different kind of buttons to satisfy various design requirements `Button`, `OmegaCenterIconButton` & `PrimarySolidButton`. Moving to material components allow us to use single `MaterialButton`. It would also allow us to use different material components. You can find more components supported by Android and how to theme them [here](https://material.io/components).
```
Comment (sanchita) [inline doc/mdc-migration.md:8]: Just one minor change here. Can you move this line from here to `Decision`?
"Moving to material components allow us to use single `MaterialButton`. It would also allow us to use different material components. You can find more components supported by Android and how to theme them [here](https://material.io/components)."

**Comment by sanchita** [comment]
LGTM!
